src/wave.ts

```diff
@@ -0,0 +1,49 @@
+import { Stage } from 'aws-cdk-lib';
+import { AddStageOpts, StageDeployment, Wave, WaveProps } from 'aws-cdk-lib/pipelines';
+import { AddGitHubStageOptions } from './github-common';
+import { GitHubWorkflow } from './pipeline';
+
+/**
+ * Multiple stages that are deployed in parallel
+ *
+ * A `Wave`, but with addition GitHub options
+ *
+ * Create with `GitHubWorkflow.addWave()` or `GitHubWorkflow.addGitHubWave()`, do not construct directly
```

Comment:
```suggestion
 * Create with `GitHubWorkflow.addWave()` or `GitHubWorkflow.addGitHubWave()`.
 * You should not have to instantiate a GitHubWave yourself.
```

This should be documented on the constructor also.